packages/remote/src/electron-node/setup/remote-setup-service.ts

```diff
@@ -91,18 +91,36 @@ export class RemoteSetupService {
             await this.copyService.copyToRemote(connection, platform, applicationZipFile);
             await this.unzipRemote(connection, platform, applicationZipFile, applicationDirectory);
         }
-        // 5. start remote backend
+        // 5. Install node_modules
+        const nodeBinDir = platform.os === OS.Type.Windows ? remoteNodeDirectory : this.scriptService.joinPath(platform, remoteNodeDirectory, 'bin');
+        const nodeExecutable = this.scriptService.joinPath(platform, nodeBinDir, 'node' + (platform.os === OS.Type.Windows ? '.exe' : ''));
+        const npmExecutable = this.scriptService.joinPath(platform, nodeBinDir, 'npm' + (platform.os === OS.Type.Windows ? '.cmd' : ''));
+
+        const nodeModulesDir = this.scriptService.joinPath(platform, libDir, 'node_modules');
+        const nodeModulesDirExists = await this.dirExistsRemote(connection, nodeModulesDir);
+        if (!nodeModulesDirExists) {
+            report('Installing node modules');
+            let prefix;
+            if (platform.os === OS.Type.Windows) {
+                // We might to switch to PowerShell beforehand on Windows
+                prefix = this.scriptService.exec(platform);
+            } else {
+                // ensure the node binary is on path so the node modules are compiled for the correct version
+                prefix = `export PATH=${nodeBinDir}:$PATH;`;
+            }
+            await connection.exec(`${prefix} cd ${libDir}; ${npmExecutable} init -y; ${npmExecutable} install node-pty tslib`);
```

Comment:
This might not work if the remote machine doesn't have a "normal" network connection. I.e. outbound HTTP traffic might be blocked to some degree.